apps/crn-server/src/data-providers/contentful/users.data-provider.ts

```diff
@@ -1,52 +1,273 @@
+    Object.entries(fields).forEach(([key, value]) => {
+      user.fields[key] = value;
+    });
+    const result = await user.update();
+    await result.publish();
   }
 }
+
+const cleanUser = (userToUpdate: UserUpdateDataObject) =>
+  Object.entries(userToUpdate).reduce((acc, [key, value]) => {
+    if (key === 'avatar') {
+      return {
+        ...acc,
+        avatar: {
+          sys: {
+            type: 'Link',
+            linkType: 'Asset',
+            id: value,
+          },
+        },
+      };
+    }
+    if (key === 'social') {
+      // the frontend only sends the fields which have values defined
+      // so need to default all social keys to null to allow unsetting
+      return {
+        ...acc,
+        website1: null,
+        website2: null,
+        twitter: null,
+        linkedIn: null,
+        github: null,
+        researcherId: null,
+        googleScholar: null,
+        researchGate: null,
+        ...(value as UserSocialLinks),
+      };
+    }
+    return { ...acc, [key]: value };
+  }, {} as { [key: string]: unknown });
+
+export const parseContentfulGraphQlUsers = (item: UserItem): UserDataObject => {
+  const normaliseArray = (
+    input: Maybe<Maybe<string>[]> | undefined,
+  ): string[] =>
+    (input || []).reduce(
+      (arr: string[], str: Maybe<string>) => (str ? [...arr, str] : arr),
+      [],
+    );
+
+  const expertiseAndResourceTags = normaliseArray(
+    item.expertiseAndResourceTags,
+  );
+  const questions = normaliseArray(item.questions);
+  const connections = normaliseArray(item.connections);
+
+  const role = item.role && isUserRole(item.role) ? item.role : 'Guest';
+  const degree =
+    item.degree && isUserDegree(item.degree) ? item.degree : undefined;
+
+  const teams: UserTeam[] = (item.teamsCollection?.items || []).reduce(
+    (userTeams: UserTeam[], team: TeamMembership | null): UserTeam[] => {
+      if (!team) {
+        return userTeams;
+      }
+
+      const { role: teamRole, inactiveSinceDate } = team;
+
+      if (!teamRole || !isTeamRole(teamRole)) {
+        return userTeams;
+      }
+
+      return [
+        ...userTeams,
+        {
+          role: teamRole,
+          inactiveSinceDate,
+          displayName: team.team?.displayName || '',
+          id: team.team?.sys.id || '',
+        },
+      ];
+    },
+    [] as UserTeam[],
+  );
+  const labs: LabResponse[] = (item.labsCollection?.items || []).reduce(
+    (userLabs: LabResponse[], lab): LabResponse[] => {
+      if (!lab) {
+        return userLabs;
+      }
+
+      return [
+        ...userLabs,
+        {
+          id: lab.sys.id,
+          name: lab.name || '',
+        },
+      ];
+    },
+    [] as LabResponse[],
+  );
+
+  return {
+    id: item.sys.id,
+    createdDate: item.sys.firstPublishedAt,
+    lastModifiedDate: item.sys.publishedAt,
+    workingGroups: [],
+    ...item,
+    email: item.email ?? '',
+    contactEmail: item.contactEmail ?? '',
```

Comment:
I see you haven't written the tests for this bit yet which might explain why this was missed but in the Squidex data provider we seem to be defaulting to undefined when Squidex returns values as Maybe (similar to Contentful). 

To be fair I can see that the test coverage for defaulting to undefined isn't great in the old data provider either.

Reply:
I had it in my head that I'd preserved the default types from the previous implementation, but apparently not. Will change to `undefined`.

Reply:
I've made the default values align across both data providers to the squidex values and added unit tests to cover both.